### PR TITLE
monitor for distributed envorinment.

### PR DESCRIPTION
**Pull request by trivialfis**
Does all printings in rank 0 worker.

No unittest for this one ... Not sure about how to do it.  Sample output:

```
[13:19:57] ======== Monitor: Learner ========
From rank: 0: 
Configure: 4e-06s, 1 calls @ 4us
GetGradient: 3.26812s, 3 calls @ 3268121us
PredictRaw: 1.88846s, 3 calls @ 1888456us
UpdateOneIter: 39.4518s, 3 calls @ 39451772us

From rank: 1: 
Configure: 5e-06s, 1 calls @ 5us
GetGradient: 3.7504s, 3 calls @ 3750402us
PredictRaw: 1.87721s, 3 calls @ 1877214us
UpdateOneIter: 39.7579s, 3 calls @ 39757925us
```

### CODE REVIEW

**Comment by trivialfis** [comment]
@chenqin Could you please help taking a look here?

File: src/learner.cc [hunk 1]
```diff
@@ -142,11 +142,13 @@ DMLC_REGISTER_PARAMETER(GenericParameter);
 class LearnerImpl : public Learner {
  public:
   explicit LearnerImpl(std::vector<std::shared_ptr<DMatrix> >  cache)
-      : configured_{false}, cache_(std::move(cache)) {}
+      : configured_{false}, cache_(std::move(cache)) {
+    monitor_.Init("Learner");
```
Comment (chenqin) [inline src/learner.cc]: LGTM, thanks for heads up!

Comment (chenqin) [reply src/learner.cc]: I wish we can keep all configurations in json

**Comment by trivialfis** [comment]
@chenqin @CodingCat @RAMitchell Thanks for the review!

File: src/learner.cc (hunk 1 above)
Comment (trivialfis) [reply src/learner.cc]: @chenqin Me too ...

**Pull request merged**